angularjs/angular.d.ts

```diff
@@ -776,31 +776,127 @@ declare module angular {
      * see https://docs.angularjs.org/api/ng/service/$filter
      */
     interface IFilterService {
+        (name: 'filter'): IFilterFilter;
+        (name: 'currency'): IFilterCurrency;
+        (name: 'number'): IFilterNumber;
+        (name: 'date'): IFilterDate;
+        (name: 'json'): IFilterJson;
+        (name: 'lowercase'): IFilterLowercase;
+        (name: 'uppercase'): IFilterUppercase;
+        (name: 'limitTo'): IFilterLimitTo;
+        (name: 'orderBy'): IFilterOrderBy;
         /**
          * Usage:
          * $filter(name);
          *
          * @param name Name of the filter function to retrieve
          */
-        (name: string): IFilterFunc;
+        <T>(name: string): T;
     }
-    
-    interface IFilterFunc {
-        <T>(array: T[], expression: string | IFilterPatternObject | IFilterPredicateFunc<T>, comparator?: IFilterComparatorFunc<T>|boolean): T[];
+        
+    interface IFilterFilter {
+        <T>(array: T[], expression: string | IFilterFilterPatternObject | IFilterFilterPredicateFunc<T>, comparator?: IFilterFilterComparatorFunc<T>|boolean): T[];
     }
     
-    interface IFilterPatternObject {
-        [name: string]: string;
+    interface IFilterFilterPatternObject {
+        [name: string]: any;
     }
     
-    interface IFilterPredicateFunc<T> {
+    interface IFilterFilterPredicateFunc<T> {
         (value: T, index: number, array: T[]): T[];
     }
     
-    interface IFilterComparatorFunc<T> {
+    interface IFilterFilterComparatorFunc<T> {
         (actual: T, expected: T): boolean;
     }
-
+    
+    interface IFilterCurrency {
+        /**
+         * Formats a number as a currency (ie $1,234.56). When no currency symbol is provided, default symbol for current locale is used.
+         * @param amount Input to filter.
+         * @param symbol Currency symbol or identifier to be displayed.
+         * @param fractionSize Number of decimal places to round the amount to, defaults to default max fraction size for current locale
+         * @return Formatted number
+         */
+        (amount: number, symbol?: string, fractionSize?: number): string;
```

Comment:
Are you sure about this? The documentation says they should be numbers. Judging from the source code (https://github.com/angular/angular.js/blob/v1.4.6/src/ng/filter/filters.js#L137), it might be possible, though it doesn't seem right calling ```Math.abs``` on a string :(